src/overlays/actors/ovl_En_Test4/z_en_test4.c

```diff
@@ -24,28 +31,557 @@ const ActorInit En_Test4_InitVars = {
+                    s32 phi_v0;
+                    u32 entranceIndex = gSaveContext.entranceIndex;
+
+                    if ((globalCtx->actorCtx.unk5 & 2)) {
+                        phi_v0 = 0xCFF;
+                    } else {
+                        phi_v0 = 0xBFF;
+                    }
+                    func_80169DCC(globalCtx, 1, entranceIndex, player->unk_3CE, phi_v0, &player->unk_3C0,
+                                  player->unk_3CC);
+
+                    if ((globalCtx->sceneNum == SCENE_TENMON_DAI) || (globalCtx->sceneNum == SCENE_00KEIKOKU)) {
+                        globalCtx->nextEntranceIndex = 0x5400;
+                    } else {
+                        globalCtx->nextEntranceIndex = 0xD800;
+                    }
+                    gSaveContext.nextCutsceneIndex = 0xFFF1;
+                    globalCtx->sceneLoadFlag = 0x14;
+                    globalCtx->unk_1887F = 2;
+                    player->stateFlags1 |= 0x200;
+                    Actor_MarkForDeath(&this->actor);
+                }
+                func_80A42198(this);
+            } else {
+                func_80A425E4(this, globalCtx);
+            }
+        }
+    }
+}
+
+void func_80A42F20(EnTest4* this, GlobalContext* globalCtx) {
+    if (!this->transitionCsTimer) {
+        if (sCurrentCs >= 0) {
+            if (ActorCutscene_GetCanPlayNext(sCurrentCs) == 0) {
+                ActorCutscene_SetIntentToPlay(sCurrentCs);
+            } else {
+                ActorCutscene_Start(sCurrentCs, &this->actor);
+                this->transitionCsTimer = 1;
+            }
+        } else {
+            this->transitionCsTimer = 1;
+        }
+    } else if (this->transitionCsTimer < 60) {
+        this->transitionCsTimer++;
+        if (this->transitionCsTimer == 10) {
+            if (this->unk_144 == 0) {
+                play_sound(NA_SE_EV_CHICKEN_CRY_M);
+            } else {
+                func_8019F128(NA_SE_EV_DOG_CRY_EVENING);
+            }
+        }
+        if (this->transitionCsTimer == 60) {
+            Player* player = PLAYER;
+
+            gSaveContext.time += CLOCK_TIME_MINUTE;
+            this->unk_146 = gSaveContext.time;
+            globalCtx->numSetupActors = -globalCtx->numSetupActors;
+            player->stateFlags1 &= ~0x200;
+        }
+    } else {
+        this->actionFunc = func_80A42AB8;
+        if (this->unk_144 == 0) {
+            this->unk_144 = 1;
+        } else {
+            this->unk_144 = 0;
+        }
+
+        if (sCurrentCs >= 0) {
+            ActorCutscene_Stop(sCurrentCs);
+        }
+        gSaveContext.unk_3F22 = 0;
+        gSaveContext.eventInf[1] &= (u8)~0x80;
+        Interface_ChangeAlpha(50);
+    }
+}
+
+void func_80A430C8(EnTest4* this, GlobalContext* globalCtx) {
+    // rain?
+    if ((CURRENT_DAY == 2) && (gSaveContext.time >= CLOCK_TIME(7, 0)) && (gSaveContext.time < CLOCK_TIME(17, 30)) &&
+        (globalCtx->envCtx.unk_F2[2] == 0)) {
+        D_801BDBB0 = 1;
```

Comment:
```suggestion
    if ((CURRENT_DAY == 2) && (gSaveContext.time >= CLOCK_TIME(7, 0)) && (gSaveContext.time < CLOCK_TIME(17, 30)) &&
        (globalCtx->envCtx.unk_F2[2] == 0)) {
        // rain?
        
        D_801BDBB0 = 1;
```
since it's referring to the contents, not the condition.